### PR TITLE
DTSSTCI-357 & DTSSTCI-262

### PR DESCRIPTION
**Pull request by tomxelliott**
### Change description ###
Create event for DSS Update Case Submissions
Configure notifications to be sent on DSS Update Case Submission

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSSTCI-357
https://tools.hmcts.net/jira/browse/DTSSTCI-262

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/sptribs/services/ccd/CaseApiService.java [hunk 1]
```diff
@@ -105,15 +105,27 @@ private CaseDataContent getCaseDataContent(String authorization, CaseData caseDa
...
                 .eventToken(getEventTokenForUpdate(authorization, userId, appsDetails.getEventIds().getUpdateEvent(),
                                                    caseId, appsDetails));
+
         } else if (eventEnum.getEventType().equalsIgnoreCase(Event.SUBMIT.getEventType())) {
+
             builder.event(uk.gov.hmcts.reform.ccd.client.model.Event.builder().id(appsDetails.getEventIds().getSubmitEvent()).build())
                 .eventToken(getEventTokenForUpdate(authorization, userId, appsDetails.getEventIds().getSubmitEvent(),
                                                    caseId, appsDetails));
+
+        } else if (eventEnum.getEventType().equalsIgnoreCase(Event.UPDATE_CASE.getEventType())) {
```
Comment (Jefex7) [inline src/main/java/uk/gov/hmcts/sptribs/services/ccd/CaseApiService.java:123]: Minor Comment : 
Sonarcloud is displaying a couple of code smells.
And a test can be added for the Event.UPDATE_CASE this small part is shown as uncovered. 
Other than that all great.

**Review by Jefex7 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/sptribs/common/notification/DssUpdateCaseSubmissionNotificationTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.sptribs.common.CommonConstants.TRIBUNAL_EMAIL_VALUE;
+import static uk.gov.hmcts.sptribs.notification.TemplateName.UPDATE_RECEIVED_CASEWORKER;
+import static uk.gov.hmcts.sptribs.notification.TemplateName.UPDATE_RECEIVED_CITIZEN;
+import static uk.gov.hmcts.sptribs.testutil.TestConstants.TEST_APPLICANT_EMAIL;
+import static uk.gov.hmcts.sptribs.testutil.TestConstants.TEST_CASE_ID;
+
+@ExtendWith(MockitoExtension.class)
+public class DssUpdateCaseSubmissionNotificationTest {
```
Comment (oleon23) [inline src/test/java/uk/gov/hmcts/sptribs/common/notification/DssUpdateCaseSubmissionNotificationTest.java:26]: Minor comment: test classes don't need to be public - is this intended?

Comment (tomxelliott) [reply src/test/java/uk/gov/hmcts/sptribs/common/notification/DssUpdateCaseSubmissionNotificationTest.java:26]: Can remove public now

**Review by oleon23 — APPROVED** [review]

**Review by gmmagruder — APPROVED** [review]

**Review by oleon23 — APPROVED** [review]

**Review by rajaram-nagarajan — APPROVED** [review]
LGTM

**Pull request merged**